### DISCUSSION TITLE
How can I teach whisper new words?

### DISCUSSION THREAD

**Discussion by MrEnigmamgine** [openai/whisper#963] [Q&A] [answered]
Hi!  I am working on a project that will use whisper to transcribe recordings of my Dungeons and Dragons games.  Since we use a lot of made up words and weird names I'd love to include a solution for training them.  For example, we often reference NPC names like Zerthimon, Vlaakith, and Mordenkainen as well as place names like Agni'hotri, Tu'narath, and Niam'd'regal.

I understand that I would need to collect labeled samples of these words. But once I have that done, how can I feed that data into Whisper?  (Preferably in a way that allows me to store the customization in a separate file and doesn't require retraining every time the app is loaded.)

Any help you guys can provide would be greatly appreciated.  I'm not expecting you to write a full tutorial, but if you could point me in the right direction I'm sure I could figure it out.

**Comment by jongwook** [comment, marked as answer]
You could make it work to some degree by prompting, by specifying `--initial_prompt "The following is a conversation during a Dungeons and Dragons game, which includes NPC names like Zerthimon, Vlaakith, and Mordenkainen as well as place names like Agni'hotri, Tu'narath, and Niam'd'regal."`, then the model will have a slightly better chance of spelling those words correctly.

If that doesn't work very reliably, the next bet is to fine-tune a Whisper model with your own examples. #364 has some pointers on how to do this.

**Reply by MrEnigmamgine** [reply]
Thank you so much!  The tutorial for fine tuning is going to be incredibly helpful, but I'll be sure to try prompting first just in case.

**Reply by chen-rn** [reply]
Hey! I'm curious how well prompting worked for you in this case?

**Reply by whc2001** [reply]
Not working really well for me, or my accent is way too crude...

![image](https://github.com/user-attachments/assets/9c3913cb-fe12-418a-b379-c61085259c01)
![image](https://github.com/user-attachments/assets/6c456445-6dff-4534-986b-c87dc5df06cf)
![image](https://github.com/user-attachments/assets/df16124b-e9fe-4b64-9649-cea1a0af8d59)

**Reply by rahulbansal16** [reply]
Does the fine-tuned model work best for you, your voice, or does it become a general model? If somebody with a different voice says the same words, will it be detected reliably?

**Reply by acidcoke** [reply]
> Not working really well for me, or my accent is way too crude...

I successfully improved the recognition in homeassistant by providing a glossary as the initial prompt e.g:
`Glossary: Lamp, Light, Screen (Right), Screen (Left), Receiver, PC, Computer, Turn, On, Off, High, Low`

**Reply by Androidmedaa** [reply]
Hi Is it possible to improve the intelligibility of some words in the Whisper Large v3 model with just a prompt? Even after cleaning the audio, it still gives an incorrect transcript during the transcription phase
